pyomo/contrib/pynumero/src/CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+cmake_minimum_required(VERSION 3.0)
+# CMake 3.0 added GIT_SUBMODULES to ExternalProject_ADD, and without it
+# the Ampl/MP checkout fails because one of the submodules (gecode) is a
+# private repository.
+
+PROJECT( pynumero )
+
+include(ExternalProject)
+
+# Targets in this project
+OPTION(BUILD_ASL "Build the PyNumero ASL interface" ON)
+OPTION(BUILD_MA27 "Build the PyNumero ma27 interface" OFF)
+OPTION(BUILD_MA57 "Build the PyNumero ma57 interface" OFF)
+
+# Dependencies that we manage / can install
+SET(AMPLMP_TAG "3.1.0" CACHE STRING
+  "AMPL/MP git tag/branch to checkout and build")
+OPTION(BUILD_AMPLMP
+  "Download and build AMPL/MP ${AMPLMP_TAG} from GitHub" OFF)
+
+# Other build / environment options
+OPTION(BUILD_AMPLMP_IF_NEEDED
+  "Automatically enable AMPLMP build if ASL not found" OFF)
+MARK_AS_ADVANCED(BUILD_AMPLMP_IF_NEEDED)
+
+#OPTION(STATIC_LINK "STATIC_LINK" OFF)
+
+# If we build AMPLMP, then we will get a dependency on dlopen
+FIND_LIBRARY(DL_LIBRARY dl)
+
+# We need the ASL and HSL libraries.  We can get them from Ipopt,
+# AMPL/MP, or ASL (netlib)
+SET(IPOPT_DIR CACHE PATH "Path to compiled Ipopt installation")
+SET(AMPLMP_DIR "" CACHE PATH "Path to compiled AMPL/MP installation")
+#SET(ASL_NETLIB_DIR "" CACHE PATH "Path to compiled ASL (netlib) installation")
+
+# Use pkg-config to get the ASL/HSL directories from the Ipopt/COIN-OR build
+FIND_PACKAGE(PkgConfig)
+IF( PKG_CONFIG_FOUND )
+  SET(_TMP "$ENV{PKG_CONFIG_PATH}")
+  SET(ENV{PKG_CONFIG_PATH} "${IPOPT_DIR}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
+  pkg_check_modules(PC_COINASL QUIET coinasl)
+  pkg_check_modules(PC_COINHSL QUIET coinhsl)
+  SET(ENV{PKG_CONFIG_PATH} "${_TMP}")
+ENDIF()
+
+# Note: the directory search order is intentional: first the modules we
+# are creating, then directories specifically set by the user, and
+# finally automatically located installations (e.g., from pkg-config)
+FIND_PATH(ASL_INCLUDE_DIR asl_pfgh.h
+  HINTS "${CMAKE_INSTALL_PREFIX}/include"
+        "${IPOPT_DIR}/include/coin/ThirdParty"
+        "${AMPLMP_DIR}/include"
```

Comment:
When I try to build using cmake with AMPL/MP built from source (following the directions in https://github.com/ampl/mp), I don't end up with a `build/include` directory as this line seems to indicate I should. Is this an AMPL/MP issue or should this hint be pointing somewhere else?

Reply:
This appears to be an issue with how I've built AMPL/MP, as the `build.py` script has no problems creating the correct include directory from the repo.

Reply:
How are you building AMPL/MP?  Are you running `make install`, and setting AMPLMP_DIR to be the same as the PREFIX (or CMAKE_INSTALL_PREFIX) for AMPL/MP?

Reply:
My problem here was not running `make install`. Doing so, I get several errors about not being able to locate the files AMPL/MP wants to put in `include`, which I can fix by manually changing some of the file locations in the `MP_INSTALL_DIR/src/asl/cmake_install.cmake` file.

Reply:
@Robbybp Are you building the master branch or the most recent release?  My work was all with the 3.1.0 release, as I was having difficulties with master, and even then I had to patch it (backporting a portability fix from fmtlib).